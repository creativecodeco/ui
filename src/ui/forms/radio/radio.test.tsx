import { fireEvent, render } from '@testing-library/react';

import type { RadioType } from '@/types';

import Radio from './radio.component';

const baseProps: RadioType = {
  name: 'test',
  value: 'test'
};

describe('<Radio />', () => {
  it('snapshot', () => {
    const { container } = render(<Radio {...baseProps} />);

    expect(container).toMatchSnapshot();
  });

  it('snapshot full props', () => {
    const { container } = render(
      <Radio
        {...baseProps}
        checked
        color='primary'
        size='lg'
        isError
        error='Error test'
        position='right'
        disabled
        label='Label'
      />
    );

    expect(container).toMatchSnapshot();
  });

  it('label', () => {
    const label = 'Hello Word';
    const { getByText } = render(<Radio {...baseProps} label={label} />);

    expect(getByText(label)).toBeInTheDocument();
  });

  it('with error', () => {
    const error = 'Error message';
    const { getByText } = render(
      <Radio {...baseProps} isError error={error} />
    );

    expect(getByText(error)).toBeInTheDocument();
  });

  it('disabled', () => {
    const { getByTestId } = render(<Radio {...baseProps} disabled />);

    expect(getByTestId('test')).toHaveProperty('disabled', true);
  });

  it('position right', () => {
    const label = 'Hello Word';
    const { getByText } = render(
      <Radio {...baseProps} label={label} position='right' />
    );

    expect(getByText(label)).toBeInTheDocument();
  });

  it('color', () => {
    const { container } = render(<Radio {...baseProps} color='primary' />);

    expect(container.querySelector('input')?.className).toEqual(
      'radio radio-color-primary radio-size-md'
    );
  });

  it('size', () => {
    const { container } = render(<Radio {...baseProps} size='lg' />);

    expect(container.querySelector('input')?.className).toEqual(
      'radio radio-size-lg'
    );
  });

  it('onChange', () => {
    const onChange = jest.fn();

    const { getByRole } = render(<Radio {...baseProps} onChange={onChange} />);

    fireEvent.click(getByRole('radio'));

    expect(onChange).toHaveBeenCalledTimes(1);
  });
});
