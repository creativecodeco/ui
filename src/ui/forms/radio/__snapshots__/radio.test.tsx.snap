// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio /> snapshot 1`] = `
<div>
  <label
    class="label justify-start gap-3 cursor-pointer"
    for="test"
  >
    <input
      class="radio radio-size-md"
      data-testid="test"
      id="test"
      name="test"
      type="radio"
    />
  </label>
</div>
`;

exports[`<Radio /> snapshot full props 1`] = `
<div>
  <label
    class="label justify-start gap-3 cursor-not-allowed"
    for="test"
  >
    <span
      class="label-text"
    >
      Label
    </span>
    <input
      checked=""
      class="radio radio-color-primary radio-size-lg"
      data-testid="test"
      disabled=""
      id="test"
      name="test"
      type="radio"
    />
    <p
      class="text-red-500"
    >
      Error test
    </p>
  </label>
</div>
`;
